app/services/homepage/fetch_tag_flares.rb

```diff
@@ -19,6 +19,8 @@ def self.articles_tag_flares(articles)
 
       flare_tag_names = ::Constants::Tags::FLARE_TAG_NAMES.to_set
       articles.pluck(:id, :cached_tag_list).map do |article_id, cached_tag_list|
+        next if cached_tag_list.blank?
```

Comment:
Interesting, is this only for articles that have no tags?

Reply:
Only for those few articles where `cached_tag_list` is nil, the code would have worked for `cached_tag_list` as an empty string, since `"".split(", ")` returns an empty array.

> probably want to look into the root cause of that nil cache_tag_list in the future but this looks like a solid workaround for now 🚀

The column can be `nil` and it looks like [they are old articles](https://dev.to/admin/blazer/queries/448-articles-with-nil-cached_tag_list) but a few of them have been recently updated. I wonder if they resurfaced in the feed somehow

Reply:
@rhymes @mstruve Any objections to adding `NOT NULL` constraints to some of these things that we expect to always have values (implied: updating the values to a sensible default)?

We allow a lot of things to be `NULL` in the DB that I often don't see `nil` checks on in code.

Reply:
No objections from me :D

Reply:
I'm game for improving this at the db level!